sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1392,9 +1392,18 @@ void CAFMaker::produce(art::Event& evt) noexcept {
   art::Handle<std::vector<raw::Trigger>> trig_handle;
   GetByLabelStrict(evt, fParams.TriggerLabel().encode(), trig_handle);
 
+  art::Handle<std::vector<raw::Trigger>> unshifted_trig_handle;
+  GetByLabelStrict(evt, fParams.UnshiftedTriggerLabel().encode(), unshifted_trig_handle);
+
+  const bool isValidTrigger = extratrig_handle.isValid() && trig_handle.isValid() && trig_handle->size() == 1;
+  const bool isValidUnshiftedTrigger = !isRealData && unshifted_trig_handle.isValid() && unshifted_trig_handle->size() == 1;
+
+  const double triggerShift = isValidUnshiftedTrigger ?
```

Comment:
Since you are using the trigger in this expression, it's reasonable to check for its validity too:
```suggestion
  const double triggerShift = (isValidUnshiftedTrigger && isValidTrigger)?
```